be/src/storage/rowset/struct_column_iterator.cpp

```diff
@@ -221,4 +240,151 @@ Status StructColumnIterator::get_row_ranges_by_zone_map(const std::vector<const
     return Status::OK();
 }
 
+Status StructColumnIterator::next_batch(size_t* n, Column* dst, ColumnAccessPath* path) {
+    if (path == nullptr || path->children().empty()) {
+        return next_batch(n, dst);
+    }
+
+    // 1. init predicate access path
+    std::vector<uint8_t> predicate_access_flags;
+    std::vector<ColumnAccessPath*> predicate_child_paths;
+    predicate_access_flags.resize(_field_iters.size(), 0);
+    predicate_child_paths.resize(_field_iters.size(), nullptr);
+    for (const auto& child : path->children()) {
+        predicate_access_flags[child->index()] = 1;
+        predicate_child_paths[child->index()] = child.get();
+    }
+
+    StructColumn* struct_column = nullptr;
+    NullColumn* null_column = nullptr;
+    if (dst->is_nullable()) {
```

Comment:
would dst be set const ? may be  a subfield in struct{struct{},struct{stuct{}}}

Reply:
the method only call once on empty chunk, so I think never be const